[wasm] Emscripten 3.1.1 bump

radekdoulik opened the pull request.

radekdoulik committed:
Initial changes for emscripten 2.0.34

radekdoulik committed:
Use emcc-link.rsp in build targets

radekdoulik committed:
Use updated docker images

Commit:
Fix compiler warnings

Co-authored-by: Zoltan Varga <vargaz@gmail.com>

radekdoulik committed:
Put `--profiling-funcs` to `_EmccLinkFlags`

radekdoulik committed:
Fix build

    src/mono/mono/mini/mini-runtime.c:3407:25: error: ‘invoke’ undeclared (first use in this function); did you mean ‘revoke’?
       3407 |                         invoke = mono_marshal_get_runtime_invoke_dynamic ();

radekdoulik committed:
Add shell to the environment

Environment setting https://github.com/emscripten-core/emscripten/blob/2.0.34/src/settings.js#L616-L641

From emscripten 2.0.25 release notes

    - Support for the 'shell' environment is now disabled by default.  Running under
      `d8`, `js`, or `jsc` is not something that most emscripten users ever want to
      do, so including the support code is, more often than not, unnecessary.  Users
      who want shell support can enable it by including 'shell' in `-s ENVIRONMENT`
      (#14535).

Example of the the size increase for bench sample:

    -a---          12/10/2021  3:35 PM         382113 dotnet.js
    -a---          12/13/2021 10:37 AM         383589 dotnet.js

radekdoulik committed:
Merge remote-tracking branch 'origin/main' into pr-emscripten-34-bump

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-34-bump

radekdoulik committed:
Add emcc-link.rsp to PlatformManifestFileEntry

radekdoulik committed:
Feedback

https://github.com/emscripten-core/emscripten/blob/2fda25eea756c78c8cb024aa5b6c2b188bf7990f/src/settings.js#L1173-L1176

-s EXPORT_ES6 is link option

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-34-bump

radekdoulik committed:
Bump emscripten version

radekdoulik committed:
Bump llvm package version and use its libclang

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-34-bump

vargaz commented on src/mono/mono/component/debugger-agent.c:
```diff
@@ -414,7 +414,7 @@ static gboolean buffer_replies;
 	tls = (DebuggerTlsData *)mono_native_tls_get_value (debugger_tls_id);
 #else
-#define GET_TLS_DATA_FROM_THREAD(thread) \
+#define GET_TLS_DATA_FROM_THREAD(...) \
```
Might be better to disable some of these new llvm warnings, the warning workarounds make the code harder to read.

radekdoulik replied:
I have added comment to this line to improve the readability. Hopefully that improves it.

radekdoulik committed:
Use newer docker images with emscripten 3.1.1

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-34-bump

radekdoulik committed:
Remove unused variable

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-311-bump

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-311-bump

radekdoulik committed:
Add note about icu repo

radekdoulik committed:
Add comment to improve code readability

lewing committed:
Bump the ICU and emsdk versions

lewing committed:
Merge branch 'main' into pr-emscripten-311-bump

lewing commented:
/azp run runtime-wasm

radical committed:
disable non-wasm builds

radical committed:
disable wasm jobs in runtime, and runtime-staging, as we'll be running runtime-wasm manually

radical commented:
/azp run runtime-wasm

radical commented:
Windows jobs are failing because of:
```
Output:
Unable to pull image mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-ltsc2022-helix-webassembly-20220118164817-dc6e6f1
Exit Code:-4
```

Linux library tests are failing for [nodejs](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-63894-merge-4d853ca4098949b2b5/WasmTestOnNodeJs-XsltCompiler.Tests/1/console.02d988d2.log?sv=2019-07-07&se=2022-03-09T01%3A01%3A11Z&sr=c&sp=rl&sig=8YaEj0Hg%2BHEfyRV90NKhWCLla%2BSppjVa5ZyLzNa6F0g%3D):
```
01:27:34] fail: if (typeof globalThis.crypto === 'undefined') {
[01:27:34] fail: ^
[01:27:34] fail:
[01:27:34] fail: ReferenceError: globalThis is not defined
```

radical commented:
Bunch of wasm.build.tests failing on nodejs due to the same error.

And `Wasm.Build.Tests.BlazorWasmBuildPublishTests.WithNativeReference_AOTOnCommandLine(config: "Debug") [FAIL]` getting killed during linking.
https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-63894-merge-8c64fb04b20e41508a/EMSDK-.WasmBuildAppTest/1/console.8fe0c164.log?sv=2019-07-07&se=2022-03-09T00%3A49%3A41Z&sr=c&sp=rl&sig=RF%2BSx%2FtoXiIuVLLEHpPys%2FMC%2FFyAQ9yXAGfKpcpVcFw%3D

radical commented:
`System.Text.Json.Tests` oom-killed during linking.

lewing commented:
Should we enable the tests that were disabled after the .34 bump? Node and json OOM are not critical path at the moment.

radical committed:
Merge remote-tracking branch 'origin/main' into pr-emscripten-311-bump

radical committed:
[wasm] Re-enable tests that were disabled due to the earlier emsdk bump

System.Linq.Tests.csproj: https://github.com/dotnet/runtime/issues/64774

System.Text.Json.Tests.csproj: https://github.com/dotnet/runtime/issues/64775

System.Text.RegularExpressions.Tests.csproj: https://github.com/dotnet/runtime/issues/64769

System.Threading.Channels.Tests.csproj: https://github.com/dotnet/runtime/issues/65012

System.Collections.Immutable.Tests.csproj: https://github.com/dotnet/runtime/issues/65335

radical committed:
[wasm] Disable nodejs tests

They are still unstable, and fail with:

```
[01:27:29] fail: /datadisks/disk1/work/A3DA08F7/w/B1B30976/e/test-main.js:97
[01:27:29] fail: if (typeof globalThis.crypto === 'undefined') {
[01:27:29] fail: ^
[01:27:29] fail:
[01:27:29] fail: ReferenceError: globalThis is not defined
[01:27:29] fail:     at Object.<anonymous> (/datadisks/disk1/work/A3DA08F7/w/B1B30976/e/test-main.js:97:1)
[01:27:29] fail:     at Module._compile (internal/modules/cjs/loader.js:778:30)
[01:27:29] fail:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
[01:27:29] fail:     at Module.load (internal/modules/cjs/loader.js:653:32)
[01:27:29] fail:     at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
[01:27:29] fail:     at Function.Module._load (internal/modules/cjs/loader.js:585:3)
[01:27:29] fail:     at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
[01:27:29] fail:     at startup (internal/bootstrap/node.js:283:19)
[01:27:29] fail:     at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
```

Issue: https://github.com/dotnet/runtime/issues/65485

radical commented:
/azp run runtime-wasm

radical committed:
fix wbt build

radical commented:
In latest run:

- AOT - [STJ oom-killed](https://helix.dot.net/api/2019-06-17/jobs/da9fc6bc-0873-476f-87c0-6c173b8d9665/workitems/normal-System.Text.Json.Tests/console)
- EAT - [STJ - Assertion at /__w/1/s/src/mono/mono/metadata/class.c:1990](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-63894-merge-fca6a02930394bb7b2/normal-System.Text.Json.Tests/1/console.e4b6d243.log?sv=2019-07-07&se=2022-03-09T04%3A14%3A06Z&sr=c&sp=rl&sig=JplMVnDD2mgSmEnj3lLogbmRgUqzOU8ksyXU3uw4poA%3D)
- regular library tests - `System.Text.RegularExpressions`
  - `chrome` - [__w/1/s/src/mono/mono/mini/../sgen/sgen-scan-object.h:91](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-63894-merge-ad333779ec0f498e8f/WasmTestOnBrowser-System.Text.RegularExpressions.Tests/1/console.5182bdbc.log?sv=2019-07-07&se=2022-03-09T04%3A10%3A36Z&sr=c&sp=rl&sig=qK%2B%2BjRAG4kqTZLw06xvvuhk%2BIrbRFEf07bdBaGS6whI%3D)
  - `v8` - [memory access out of bounds](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-63894-merge-7d0afab01325420e96/normal-System.Text.RegularExpressions.Tests/1/console.9248a90d.log?sv=2019-07-07&se=2022-03-09T04%3A10%3A36Z&sr=c&sp=rl&sig=IcFpsg5uYRD1jX991YlbIMcYzD2YsiefAv%2BbMZErFZw%3D)

- WBT failed to compile, due to my patch

radical commented:
/azp run runtime-wasm

maraf committed:
Update path to node in helix commands.

radekdoulik commented:
/azp run runtime-wasm

radekdoulik commented:
Looks like the memory issue is still present, just occurring in different tests now. I have opened https://github.com/dotnet/runtime/pull/65496 draft to see whether going back to older emscripten would help.

radical committed:
Merge remote-tracking branch 'origin/main' into pr-emscripten-311-bump

radical committed:
disable non-wasm builds

radical commented:
/azp run runtime-wasm

radical committed:
Merge remote-tracking branch 'origin/main' into pr-emscripten-311-bump

radical committed:
Merge remote-tracking branch 'origin/main' into pr-emscripten-311-bump

radical committed:
[wasm] Make emscripten version mismatch an error on CI

radical committed:
Bump emscripten packages used to preview3, which has emsdk 3.1.1

radical commented:
Updated the emscripten packages being used to match the bump. Without this WBT used emsdk 2.0.23 .

radical commented:
/azp run runtime-wasm

radical committed:
Update emscripten package version in Versions.props tooo

radical commented:
/azp run runtime-wasm

radical commented:
Just for info: `main` is still using an older version of the sdk for WBT workload tests. That version uses emscripten 2.0.23 .

[This](https://github.com/dotnet/runtime/pull/65003) bumps the SDK Version, and updates the tests.

radical committed:
Re-enable wasm jobs

radical committed:
Merge remote-tracking branch 'origin/main' into pr-emscripten-311-bump

radical committed:
enable non-wasm jobs

radekdoulik committed:
Add more missing exports

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-311-bump

radekdoulik commented:
/azp run runtime-wasm

radekdoulik committed:
__dl_seterr shouldn't be needed

radekdoulik commented:
/azp run runtime-wasm

radekdoulik commented:
/azp run runtime-wasm

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-311-bump

radekdoulik commented:
/azp run runtime-wasm

radekdoulik committed:
Fix MicrosoftNETCoreRuntimeICUTransportVersion

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-311-bump

radekdoulik committed:
Put back the 3.1.1 docker images after merge

radekdoulik committed:
Revert "fix wbt build"

This reverts commit f7c77d995e36130118dd9d98c2c66809aaf42fe2.

radekdoulik committed:
Revert "[wasm] Disable nodejs tests"

This reverts commit 7b90965638b7e90bf4d35c69696f2f361a04ef20.

radekdoulik committed:
Use newer ICU package, last that still had 3.1.1

radekdoulik commented:
/azp run runtime-wasm

radekdoulik committed:
Try different helix queue parameter

radekdoulik committed:
Revert "Try different helix queue parameter"

This reverts commit a34230f7fe04760df30c5e63e14826575f1a7545.

radekdoulik committed:
Fix helix queue with docker tag

radekdoulik commented:
/azp run runtime-wasm

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-311-bump

radekdoulik commented:
/azp run runtime-wasm

radekdoulik commented:
The windows libraries AOT is failing with `ImportError: DLL load failed while importing _ctypes: The specified module could not be found.`. That's probably same as https://github.com/dotnet/runtime/issues/65859

The linux libraries AOT is OOM.

The WBT is probably OOM too.
```
[] emcc : error : '/datadisks/disk1/work/B00009BF/p/dotnet-workload/packs/Microsoft.NET.Runtime.Emscripten.3.1.1.Sdk.linux-x64/7.0.0-preview.3.22121.1/tools/bin/wasm-emscripten-finalize --minimize-wasm-changes -g --dyncalls-i64 --dwarf /datadisks/disk1/work/B00009BF/w/ACA3095D/e/blz_deploy_on_build_Debug_False/obj/Debug/net7.0/wasm/for-publish/dotnet.wasm -o /datadisks/disk1/work/B00009BF/w/ACA3095D/e/blz_deploy_on_build_Debug_False/obj/Debug/net7.0/wasm/for-publish/dotnet.wasm --detect-features' failed (received SIGKILL (-9)) [/datadisks/disk1/work/B00009BF/w/ACA3095D/e/blz_deploy_on_build_Debug_False/blz_deploy_on_build_Debug_False.csproj]
[] /datadisks/disk1/work/B00009BF/p/dotnet-workload/packs/Microsoft.NET.Runtime.WebAssembly.Sdk/7.0.0-ci/Sdk/WasmApp.Native.targets(419,5): error MSB3073: The command "emcc "@/datadisks/disk1/work/B00009BF/p/dotnet-workload/packs/Microsoft.NETCore.App.Runtime.Mono.browser-wasm/7.0.0-ci/runtimes/browser-wasm/native/src/emcc-default.rsp" "@/datadisks/disk1/work/B00009BF/p/dotnet-workload/packs/Microsoft.NETCore.App.Runtime.Mono.browser-wasm/7.0.0-ci/runtimes/browser-wasm/native/src/emcc-link.rsp" "@/datadisks/disk1/work/B00009BF/w/ACA3095D/e/blz_deploy_on_build_Debug_False/obj/Debug/net7.0/wasm/for-publish/emcc-link.rsp"" exited with code 1. [/datadisks/disk1/work/B00009BF/w/ACA3095D/e/blz_deploy_on_build_Debug_False/blz_deploy_on_build_Debug_False.csproj]
[] 
[] Build FAILED.
```
```

radekdoulik committed:
Try to set PYTHONPATH for emscripten

radical committed:
[wasm] Do not set _WasmDevel=true for blazor projects

radekdoulik committed:
Increase timeout

The AOT tests are canceled after timeout

radekdoulik committed:
Fix WBT tests

Co-authored-by: Ankit Jain <radical@gmail.com>

radical commented:
/azp run runtime-wasm

radical commented:
`Microsoft.Extensions.Logging.Tests` windows/aot/v8 failed:

```
Starting:    Microsoft.Extensions.Logging.Tests.dll
Message1
:
testKey3:testValue, testKey2:, testKey1:
missing function: ntohs
Error
    at Object.onAbort (test-main.js:176:60)
    at abort (./dotnet.js:1316:24)
    at _ntohs (./dotnet.js:4990:35)
    at readSockaddr (./dotnet.js:5091:18)
    at getSocketAddress (./dotnet.js:5122:18)
    at ___syscall_connect (./dotnet.js:5153:18)
    at connect (wasm://wasm/0b9a6656:wasm-function[97015]:0x1494776)
    at __openlog (wasm://wasm/0b9a6656:wasm-function[96682]:0x1487db2)
    at _vsyslog (wasm://wasm/0b9a6656:wasm-function[96685]:0x1487e74)
    at __vsyslog (wasm://wasm/0b9a6656:wasm-function[96684]:0x1487e23)
    at syslog (wasm://wasm/0b9a6656:wasm-function[96683]:0x1487dd3)
    at SystemNative_SysLog (wasm://wasm/0b9a6656:wasm-function[96406]:0x147decb)
    at aot_wrapper_pinvoke_corlib__Interop_sl_Sys___le_SysLog_gt_g____PInvoke___verbar_116_0_pinvoke_void_cl1e_Interop_2fSys_2fSysLogPriority_cl7_byte_2a_cl7_byte_2a_void_cl1e_Interop_2fSys_2fSysLogPriority_cl7_byte_2a_cl7_byte_2a_ (wasm://wasm/0b9a6656:wasm-function[37040]:0x9ce783)
    at corlib_Interop_Sys_SysLog_Interop_Sys_SysLogPriority_string_string (wasm://wasm/0b9a6656:wasm-function[37039]:0x9ce6d4)
    at corlib_System_Diagnostics_DebugProvider_WriteToDebugger_string (wasm://wasm/0b9a6656:wasm-function[48914]:0xbf8dea)
    at corlib_System_Diagnostics_DebugProvider_WriteCore_string (wasm://wasm/0b9a6656:wasm-function[48908]:0xbf8957)
    at System_Diagnostics_TraceSource_System_Diagnostics_DefaultTraceListener_Write_string_bool (wasm://wasm/0b9a6656:wasm-function[20098]:0x48f537)
    at System_Diagnostics_TraceSource_System_Diagnostics_DefaultTraceListener_Write_string (wasm://wasm/0b9a6656:wasm-function[20097]:0x48f469)
    at System_Diagnostics_TraceSource_System_Diagnostics_TraceListener_WriteHeader_string_System_Diagnostics_TraceEventType_int (wasm://wasm/0b9a6656:wasm-function[20171]:0x494a54)
    at System_Diagnostics_TraceSource_System_Diagnostics_TraceListener_TraceEvent_System_Diagnostics_TraceEventCache_string_System_Diagnostics_TraceEventType_int_string (wasm://wasm/0b9a6656:wasm-function[20173]:0x4958b4)
    at System_Diagnostics_TraceSource_System_Diagnostics_TraceSource_TraceEvent_System_Diagnostics_TraceEventType_int_string (wasm://wasm/0b9a6656:wasm-function[20199]:0x4979ed)
    at Microsoft_Extensions_Logging_TraceSource_Microsoft_Extensions_Logging_TraceSource_TraceSourceLogger_Log_TState_REF_Microsoft_Extensions_Logging_LogLevel_Microsoft_Extensions_Logging_EventId_TState_REF_System_Exception_System_Func_3_TState_REF_System_Exception_string (wasm://wasm/0b9a6656:wasm-function[14659]:0x37e40f)
    at invoke_viiiiiii (./dotnet.js:8309:27)
    at Microsoft_Extensions_Logging_Microsoft_Extensions_Logging_Logger__Logg__LoggerLog_12_0_TState_REF_Microsoft_Extensions_Logging_LogLevel_Microsoft_Extensions_Logging_EventId_Microsoft_Extensions_Logging_ILogger_System_Exception_System_Func_3_TState_REF_System_Exception_string_System_Collections_Generic_List_1_System_Exception__TState_REF_ (wasm://wasm/0b9a6656:wasm-function[11664]:0x2a88a7)
    at Microsoft_Extensions_Logging_Microsoft_Extensions_Logging_Logger_Log_TState_REF_Microsoft_Extensions_Logging_LogLevel_Microsoft_Extensions_Logging_EventId_TState_REF_System_Exception_System_Func_3_TState_REF_System_Exception_string (wasm://wasm/0b9a6656:wasm-function[11654]:0x2a78c0)
    at Microsoft_Extensions_Logging_Tests_Microsoft_Extensions_Logging_Test_TraceSourceLoggerTest_Log_Shoud_Add_Exception_To_Message_Whether_Formatter_Is_Null_Or_Not_bool (wasm://wasm/0b9a6656:wasm-function[13246]:0x31a8af)
    at System_Diagnostics_TraceSource_aot_wrapper_gsharedvt_out_sig_void_u1 (wasm://wasm/0b9a6656:wasm-function[20319]:0x49cbfc)
    at jit_call_cb (wasm://wasm/0b9a6656:wasm-function[78848]:0x1282606)
    at invoke_vi (./dotnet.js:8199:27)
    at mono_llvm_cpp_catch_exception (wasm://wasm/0b9a6656:wasm-function[79279]:0x12a101b)
    at do_jit_call (wasm://wasm/0b9a6656:wasm-function[78795]:0x128140b)
    at interp_exec_method (wasm://wasm/0b9a6656:wasm-function[78723]:0x1273b58)
    at interp_runtime_invoke (wasm://wasm/0b9a6656:wasm-function[78722]:0x1272649)
    at mono_jit_runtime_invoke (wasm://wasm/0b9a6656:wasm-function[95083]:0x1455382)
    at do_runtime_invoke (wasm://wasm/0b9a6656:wasm-function[83692]:0x1312f55)
    at mono_runtime_invoke_checked (wasm://wasm/0b9a6656:wasm-function[83690]:0x1312efd)
    at mono_runtime_try_invoke_span (wasm://wasm/0b9a6656:wasm-function[83851]:0x1317573)
    at mono_runtime_invoke_span_checked (wasm://wasm/0b9a6656:wasm-function[83852]:0x1317760)
    at ves_icall_InternalInvoke (wasm://wasm/0b9a6656:wasm-function[82489]:0x12f0d8c)
    at ves_icall_InternalInvoke_raw (wasm://wasm/0b9a6656:wasm-function[82645]:0x12f5def)
    at aot_wrapper_corlib_System_dot_Reflection_System_dot_Reflection_dot_RuntimeMethodInfo__InternalInvoke_pinvoke_obj_this_objbcls17_Span_601_3cobject_3e_26__attrs_1bclsc_Exception_26__attrs_2obj_cls1d_Reflection_dRuntimeMethodInfo_objbcls17_Span_601_3cobject_3e_26__attrs_1bclsc_Exception_26__attrs_2 (wasm://wasm/0b9a6656:wasm-function[46555]:0xb80e63)
    at invoke_iiiiii (./dotnet.js:8210:34)
    at corlib_System_Reflection_RuntimeMethodInfo_InvokeWorker_object_System_Reflection_BindingFlags_System_Span_1_object (wasm://wasm/0b9a6656:wasm-function[46556]:0xb8100b)
    at corlib_System_Reflection_RuntimeMethodInfo_Invoke_object_System_Reflection_BindingFlags_System_Reflection_Binder_object___System_Globalization_CultureInfo (wasm://wasm/0b9a6656:wasm-function[46584]:0xb83677)
    at xunit_execution_dotnet_Xunit_Sdk_TestInvoker_1_TTestCase_REF_CallTestMethod_object (wasm://wasm/0b9a6656:wasm-function[71274]:0x10b0777)
    at invoke_iiii (./dotnet.js:8155:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestInvoker_1__c__DisplayClass48_0___InvokeTestMethodAsyncb__1d_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71292]:0x10b4177)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_TStateMachine_INST_TStateMachine_INST__6 (wasm://wasm/0b9a6656:wasm-function[73859]:0x11604f1)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncTaskMethodBuilder_Start_TStateMachine_INST_TStateMachine_INST_ (wasm://wasm/0b9a6656:wasm-function[73227]:0x11455c2)
    at xunit_execution_dotnet_Xunit_Sdk_TestInvoker_1__c__DisplayClass48_0_TTestCase_REF__InvokeTestMethodAsyncb__1 (wasm://wasm/0b9a6656:wasm-function[71291]:0x10b3a32)
    at corlib_wrapper_delegate_invoke_System_Func_1_System_Threading_Tasks_Task_invoke_TResult (wasm://wasm/0b9a6656:wasm-function[51356]:0xc87ec0)
    at invoke_iii (./dotnet.js:8122:34)
    at xunit_execution_dotnet_Xunit_Sdk_ExecutionTimer__AggregateAsyncd__4_MoveNext (wasm://wasm/0b9a6656:wasm-function[71106]:0x1099eaf)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Xunit_Sdk_ExecutionTimer__AggregateAsyncd__4_Xunit_Sdk_ExecutionTimer__AggregateAsyncd__4_ (wasm://wasm/0b9a6656:wasm-function[73172]:0x1143690)
    at xunit_execution_dotnet_Xunit_Sdk_ExecutionTimer_AggregateAsync_System_Func_1_System_Threading_Tasks_Task (wasm://wasm/0b9a6656:wasm-function[71104]:0x1099bde)
    at xunit_execution_dotnet_Xunit_Sdk_TestInvoker_1__c__DisplayClass48_0_TTestCase_REF__InvokeTestMethodAsyncb__0 (wasm://wasm/0b9a6656:wasm-function[71290]:0x10b38e0)
    at corlib_wrapper_delegate_invoke_System_Func_1_System_Threading_Tasks_Task_invoke_TResult (wasm://wasm/0b9a6656:wasm-function[51356]:0xc87ec0)
    at invoke_iii (./dotnet.js:8122:34)
    at xunit_core_Xunit_Sdk_ExceptionAggregator__RunAsyncd__9_MoveNext (wasm://wasm/0b9a6656:wasm-function[70181]:0x105e41e)
    at xunit_core_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Xunit_Sdk_ExceptionAggregator__RunAsyncd__9_Xunit_Sdk_ExceptionAggregator__RunAsyncd__9_ (wasm://wasm/0b9a6656:wasm-function[70439]:0x1072cec)
    at xunit_core_Xunit_Sdk_ExceptionAggregator_RunAsync_System_Func_1_System_Threading_Tasks_Task (wasm://wasm/0b9a6656:wasm-function[70174]:0x105d387)
    at invoke_iiii (./dotnet.js:8155:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestInvoker_1__InvokeTestMethodAsyncd__48_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71296]:0x10b584c)
    at corlib_aot_wrapper_gsharedvt_out_sig_pinvoke_void_this_ (wasm://wasm/0b9a6656:wasm-function[53092]:0xced5c6)
    at jit_call_cb (wasm://wasm/0b9a6656:wasm-function[78848]:0x1282606)
    at invoke_vi (./dotnet.js:8199:27)
    at mono_llvm_cpp_catch_exception (wasm://wasm/0b9a6656:wasm-function[79279]:0x12a101b)
    at do_jit_call (wasm://wasm/0b9a6656:wasm-function[78795]:0x128140b)
    at interp_exec_method (wasm://wasm/0b9a6656:wasm-function[78723]:0x1273b58)
    at interp_entry (wasm://wasm/0b9a6656:wasm-function[78776]:0x1280838)
    at interp_entry_instance_1 (wasm://wasm/0b9a6656:wasm-function[78862]:0x1282adf)
    at corlib_aot_wrapper_gsharedvt_in_sig_void_this_bii (wasm://wasm/0b9a6656:wasm-function[55040]:0xd12736)
    at xunit_execution_dotnet_Xunit_Sdk_TestInvoker_1_TTestCase_REF_InvokeTestMethodAsync_object (wasm://wasm/0b9a6656:wasm-function[71277]:0x10b0a46)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestInvoker_InvokeTestMethodAsync_object (wasm://wasm/0b9a6656:wasm-function[71488]:0x10d151f)
    at invoke_iiii (./dotnet.js:8155:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestInvoker_1___RunAsyncb__47_0d_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71280]:0x10b206f)
    at corlib_aot_wrapper_gsharedvt_out_sig_pinvoke_void_this_ (wasm://wasm/0b9a6656:wasm-function[53092]:0xced5c6)
    at jit_call_cb (wasm://wasm/0b9a6656:wasm-function[78848]:0x1282606)
    at invoke_vi (./dotnet.js:8199:27)
    at mono_llvm_cpp_catch_exception (wasm://wasm/0b9a6656:wasm-function[79279]:0x12a101b)
    at do_jit_call (wasm://wasm/0b9a6656:wasm-function[78795]:0x128140b)
    at interp_exec_method (wasm://wasm/0b9a6656:wasm-function[78723]:0x1273b58)
    at interp_entry (wasm://wasm/0b9a6656:wasm-function[78776]:0x1280838)
    at interp_entry_instance_1 (wasm://wasm/0b9a6656:wasm-function[78862]:0x1282adf)
    at corlib_aot_wrapper_gsharedvt_in_sig_void_this_bii (wasm://wasm/0b9a6656:wasm-function[55040]:0xd12736)
    at xunit_execution_dotnet_Xunit_Sdk_TestInvoker_1_TTestCase_REF__RunAsyncb__47_0 (wasm://wasm/0b9a6656:wasm-function[71279]:0x10b0be4)
    at xunit_execution_dotnet_wrapper_delegate_invoke_System_Func_1_System_Threading_Tasks_Task_1_System_Decimal_invoke_TResult (wasm://wasm/0b9a6656:wasm-function[72578]:0x1116a40)
    at invoke_iii (./dotnet.js:8122:34)
    at xunit_execution_dotnet_Xunit_Sdk_ExceptionAggregator__RunAsyncd__10_1_System_Decimal_MoveNext (wasm://wasm/0b9a6656:wasm-function[73221]:0x114514c)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Xunit_Sdk_ExceptionAggregator__RunAsyncd__10_1_System_Decimal_Xunit_Sdk_ExceptionAggregator__RunAsyncd__10_1_System_Decimal_ (wasm://wasm/0b9a6656:wasm-function[73856]:0x1160018)
    at xunit_execution_dotnet_Xunit_Sdk_ExceptionAggregator_RunAsync_System_Decimal_System_Func_1_System_Threading_Tasks_Task_1_System_Decimal (wasm://wasm/0b9a6656:wasm-function[73220]:0x1144e7e)
    at xunit_execution_dotnet_Xunit_Sdk_TestInvoker_1_TTestCase_REF_RunAsync (wasm://wasm/0b9a6656:wasm-function[71276]:0x10b091c)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestRunner_InvokeTestMethodAsync_Xunit_Sdk_ExceptionAggregator (wasm://wasm/0b9a6656:wasm-function[71504]:0x10d28b5)
    at invoke_iiii (./dotnet.js:8155:34)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestRunner__InvokeTestAsyncd__4_MoveNext (wasm://wasm/0b9a6656:wasm-function[71505]:0x10d2f3f)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Xunit_Sdk_XunitTestRunner__InvokeTestAsyncd__4_Xunit_Sdk_XunitTestRunner__InvokeTestAsyncd__4_ (wasm://wasm/0b9a6656:wasm-function[73286]:0x1148211)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestRunner_InvokeTestAsync_Xunit_Sdk_ExceptionAggregator (wasm://wasm/0b9a6656:wasm-function[71503]:0x10d2773)
    at xunit_execution_dotnet_Xunit_Sdk_TestRunner_1__c__DisplayClass43_0_TTestCase_REF__RunAsyncb__0 (wasm://wasm/0b9a6656:wasm-function[71351]:0x10ba1f8)
    at xunit_execution_dotnet_wrapper_delegate_invoke_System_Func_1_System_Threading_Tasks_Task_1_System_Tuple_2_System_Decimal_string_invoke_TResult (wasm://wasm/0b9a6656:wasm-function[72624]:0x1118309)
    at invoke_iii (./dotnet.js:8122:34)
    at xunit_core_Xunit_Sdk_ExceptionAggregator__RunAsyncd__10_1_T_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[70177]:0x105d977)
    at xunit_core_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_TStateMachine_INST_TStateMachine_INST_ (wasm://wasm/0b9a6656:wasm-function[70608]:0x10774e1)
    at xunit_core_System_Runtime_CompilerServices_AsyncTaskMethodBuilder_1_TResult_REF_Start_TStateMachine_INST_TStateMachine_INST_ (wasm://wasm/0b9a6656:wasm-function[70442]:0x1072fea)
    at xunit_core_Xunit_Sdk_ExceptionAggregator_RunAsync_T_REF_System_Func_1_System_Threading_Tasks_Task_1_T_REF (wasm://wasm/0b9a6656:wasm-function[70175]:0x105d4f0)
    at invoke_iiii (./dotnet.js:8155:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestRunner_1__RunAsyncd__43_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71352]:0x10bb32a)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_TStateMachine_INST_TStateMachine_INST__9 (wasm://wasm/0b9a6656:wasm-function[73864]:0x1161372)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncTaskMethodBuilder_1_TResult_REF_Start_TStateMachine_INST_TStateMachine_INST__8 (wasm://wasm/0b9a6656:wasm-function[73241]:0x1145a82)
    at xunit_execution_dotnet_Xunit_Sdk_TestRunner_1_TTestCase_REF_RunAsync (wasm://wasm/0b9a6656:wasm-function[71349]:0x10ba112)
    at invoke_iii (./dotnet.js:8122:34)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTheoryTestCaseRunner__RunTestAsyncd__11_MoveNext (wasm://wasm/0b9a6656:wasm-function[71532]:0x10db4d1)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Xunit_Sdk_XunitTheoryTestCaseRunner__RunTestAsyncd__11_Xunit_Sdk_XunitTheoryTestCaseRunner__RunTestAsyncd__11_ (wasm://wasm/0b9a6656:wasm-function[73292]:0x1148944)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTheoryTestCaseRunner_RunTestAsync (wasm://wasm/0b9a6656:wasm-function[71515]:0x10d3d2b)
    at invoke_iii (./dotnet.js:8122:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestCaseRunner_1__RunAsyncd__19_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71173]:0x10a2c78)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_TStateMachine_INST_TStateMachine_INST__1 (wasm://wasm/0b9a6656:wasm-function[73846]:0x115e824)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncTaskMethodBuilder_1_TResult_REF_Start_TStateMachine_INST_TStateMachine_INST__1 (wasm://wasm/0b9a6656:wasm-function[73190]:0x1143e81)
    at xunit_execution_dotnet_Xunit_Sdk_TestCaseRunner_1_TTestCase_REF_RunAsync (wasm://wasm/0b9a6656:wasm-function[71098]:0x1099661)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTheoryTestCase_RunAsync_Xunit_Abstractions_IMessageSink_Xunit_Sdk_IMessageBus_object___Xunit_Sdk_ExceptionAggregator_System_Threading_CancellationTokenSource (wasm://wasm/0b9a6656:wasm-function[71769]:0x10efc12)
    at Microsoft_DotNet_XUnitExtensions_Microsoft_DotNet_XUnitExtensions_SkippedTheoryTestCase__n__0_Xunit_Abstractions_IMessageSink_Xunit_Sdk_IMessageBus_object___Xunit_Sdk_ExceptionAggregator_System_Threading_CancellationTokenSource (wasm://wasm/0b9a6656:wasm-function[4138]:0x1595b2)
    at invoke_iiiiiiii (./dotnet.js:8298:34)
    at Microsoft_DotNet_XUnitExtensions_Microsoft_DotNet_XUnitExtensions_SkippedTheoryTestCase__RunAsyncd__2_MoveNext (wasm://wasm/0b9a6656:wasm-function[4139]:0x1599b4)
    at Microsoft_DotNet_XUnitExtensions_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Microsoft_DotNet_XUnitExtensions_SkippedTheoryTestCase__RunAsyncd__2_Microsoft_DotNet_XUnitExtensions_SkippedTheoryTestCase__RunAsyncd__2_ (wasm://wasm/0b9a6656:wasm-function[4308]:0x15ff6b)
    at Microsoft_DotNet_XUnitExtensions_Microsoft_DotNet_XUnitExtensions_SkippedTheoryTestCase_RunAsync_Xunit_Abstractions_IMessageSink_Xunit_Sdk_IMessageBus_object___Xunit_Sdk_ExceptionAggregator_System_Threading_CancellationTokenSource (wasm://wasm/0b9a6656:wasm-function[4137]:0x1594dc)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestMethodRunner_RunTestCaseAsync_Xunit_Sdk_IXunitTestCase (wasm://wasm/0b9a6656:wasm-function[71501]:0x10d2682)
    at invoke_iiii (./dotnet.js:8155:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestMethodRunner_1__RunTestCasesAsyncd__32_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71324]:0x10b8fc6)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_TStateMachine_INST_TStateMachine_INST__8 (wasm://wasm/0b9a6656:wasm-function[73861]:0x1160d52)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncTaskMethodBuilder_1_TResult_REF_Start_TStateMachine_INST_TStateMachine_INST__7 (wasm://wasm/0b9a6656:wasm-function[73234]:0x11457fc)
    at xunit_execution_dotnet_Xunit_Sdk_TestMethodRunner_1_TTestCase_REF_RunTestCasesAsync (wasm://wasm/0b9a6656:wasm-function[71316]:0x10b67db)
    at invoke_iii (./dotnet.js:8122:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestMethodRunner_1__RunAsyncd__31_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71317]:0x10b6fde)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_TStateMachine_INST_TStateMachine_INST__7 (wasm://wasm/0b9a6656:wasm-function[73860]:0x116095c)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncTaskMethodBuilder_1_TResult_REF_Start_TStateMachine_INST_TStateMachine_INST__6 (wasm://wasm/0b9a6656:wasm-function[73232]:0x114575f)
    at xunit_execution_dotnet_Xunit_Sdk_TestMethodRunner_1_TTestCase_REF_RunAsync (wasm://wasm/0b9a6656:wasm-function[71315]:0x10b6647)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestClassRunner_RunTestMethodAsync_Xunit_Abstractions_ITestMethod_Xunit_Abstractions_IReflectionMethodInfo_System_Collections_Generic_IEnumerable_1_Xunit_Sdk_IXunitTestCase_object__ (wasm://wasm/0b9a6656:wasm-function[71419]:0x10c51fa)
    at invoke_iiiiiii (./dotnet.js:8243:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestClassRunner_1__RunTestMethodsAsyncd__38_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71217]:0x10aa566)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_TStateMachine_INST_TStateMachine_INST__3 (wasm://wasm/0b9a6656:wasm-function[73849]:0x115f10b)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncTaskMethodBuilder_1_TResult_REF_Start_TStateMachine_INST_TStateMachine_INST__3 (wasm://wasm/0b9a6656:wasm-function[73199]:0x11441c2)
    at xunit_execution_dotnet_Xunit_Sdk_TestClassRunner_1_TTestCase_REF_RunTestMethodsAsync (wasm://wasm/0b9a6656:wasm-function[71202]:0x10a5093)
    at invoke_iii (./dotnet.js:8122:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestClassRunner_1__RunAsyncd__37_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71210]:0x10a6672)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_TStateMachine_INST_TStateMachine_INST__2 (wasm://wasm/0b9a6656:wasm-function[73848]:0x115ed16)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncTaskMethodBuilder_1_TResult_REF_Start_TStateMachine_INST_TStateMachine_INST__2 (wasm://wasm/0b9a6656:wasm-function[73197]:0x1144125)
    at xunit_execution_dotnet_Xunit_Sdk_TestClassRunner_1_TTestCase_REF_RunAsync (wasm://wasm/0b9a6656:wasm-function[71201]:0x10a4eff)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestCollectionRunner_RunTestClassAsync_Xunit_Abstractions_ITestClass_Xunit_Abstractions_IReflectionTypeInfo_System_Collections_Generic_IEnumerable_1_Xunit_Sdk_IXunitTestCase (wasm://wasm/0b9a6656:wasm-function[71458]:0x10cc9ab)
    at invoke_iiiiii (./dotnet.js:8210:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestCollectionRunner_1__RunTestClassesAsyncd__28_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71248]:0x10af332)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_TStateMachine_INST_TStateMachine_INST__5 (wasm://wasm/0b9a6656:wasm-function[73853]:0x115fb08)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncTaskMethodBuilder_1_TResult_REF_Start_TStateMachine_INST_TStateMachine_INST__5 (wasm://wasm/0b9a6656:wasm-function[73213]:0x1144be1)
    at xunit_execution_dotnet_Xunit_Sdk_TestCollectionRunner_1_TTestCase_REF_RunTestClassesAsync (wasm://wasm/0b9a6656:wasm-function[71237]:0x10ab6b6)
    at invoke_iii (./dotnet.js:8122:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestCollectionRunner_1__RunAsyncd__27_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71241]:0x10ac4e1)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_TStateMachine_INST_TStateMachine_INST__4 (wasm://wasm/0b9a6656:wasm-function[73852]:0x115f713)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncTaskMethodBuilder_1_TResult_REF_Start_TStateMachine_INST_TStateMachine_INST__4 (wasm://wasm/0b9a6656:wasm-function[73211]:0x1144b44)
    at xunit_execution_dotnet_Xunit_Sdk_TestCollectionRunner_1_TTestCase_REF_RunAsync (wasm://wasm/0b9a6656:wasm-function[71236]:0x10ab522)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestAssemblyRunner_RunTestCollectionAsync_Xunit_Sdk_IMessageBus_Xunit_Abstractions_ITestCollection_System_Collections_Generic_IEnumerable_1_Xunit_Sdk_IXunitTestCase_System_Threading_CancellationTokenSource (wasm://wasm/0b9a6656:wasm-function[71373]:0x10bf0d2)
    at invoke_iiiiiii (./dotnet.js:8243:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestAssemblyRunner_1__RunTestCollectionsAsyncd__42_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71159]:0x10a166c)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_TStateMachine_INST_TStateMachine_INST__0 (wasm://wasm/0b9a6656:wasm-function[73843]:0x115e213)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncTaskMethodBuilder_1_TResult_REF_Start_TStateMachine_INST_TStateMachine_INST__0 (wasm://wasm/0b9a6656:wasm-function[73182]:0x1143bd4)
    at xunit_execution_dotnet_Xunit_Sdk_TestAssemblyRunner_1_TTestCase_REF_RunTestCollectionsAsync_Xunit_Sdk_IMessageBus_System_Threading_CancellationTokenSource (wasm://wasm/0b9a6656:wasm-function[71144]:0x109d7c2)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestAssemblyRunner__n__0_Xunit_Sdk_IMessageBus_System_Threading_CancellationTokenSource (wasm://wasm/0b9a6656:wasm-function[71375]:0x10bf200)
    at invoke_iiiii (./dotnet.js:8254:34)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestAssemblyRunner__RunTestCollectionsAsyncd__14_MoveNext (wasm://wasm/0b9a6656:wasm-function[71387]:0x10c00e1)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Xunit_Sdk_XunitTestAssemblyRunner__RunTestCollectionsAsyncd__14_Xunit_Sdk_XunitTestAssemblyRunner__RunTestCollectionsAsyncd__14_ (wasm://wasm/0b9a6656:wasm-function[73247]:0x1145d6f)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestAssemblyRunner_RunTestCollectionsAsync_Xunit_Sdk_IMessageBus_System_Threading_CancellationTokenSource (wasm://wasm/0b9a6656:wasm-function[71372]:0x10bef94)
    at invoke_iiiii (./dotnet.js:8254:34)
    at xunit_execution_dotnet_Xunit_Sdk_TestAssemblyRunner_1__RunAsyncd__41_TTestCase_REF_MoveNext (wasm://wasm/0b9a6656:wasm-function[71152]:0x109ed07)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_TStateMachine_INST_TStateMachine_INST_ (wasm://wasm/0b9a6656:wasm-function[73842]:0x115dde9)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncTaskMethodBuilder_1_TResult_REF_Start_TStateMachine_INST_TStateMachine_INST_ (wasm://wasm/0b9a6656:wasm-function[73180]:0x1143b37)
    at xunit_execution_dotnet_Xunit_Sdk_TestAssemblyRunner_1_TTestCase_REF_RunAsync (wasm://wasm/0b9a6656:wasm-function[71143]:0x109d5e2)
    at invoke_iii (./dotnet.js:8122:34)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestFrameworkExecutor__RunTestCasesd__8_MoveNext (wasm://wasm/0b9a6656:wasm-function[71764]:0x10ef37a)
    at xunit_execution_dotnet_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Xunit_Sdk_XunitTestFrameworkExecutor__RunTestCasesd__8_Xunit_Sdk_XunitTestFrameworkExecutor__RunTestCasesd__8_ (wasm://wasm/0b9a6656:wasm-function[73331]:0x1149b1a)
    at xunit_execution_dotnet_Xunit_Sdk_XunitTestFrameworkExecutor_RunTestCases_System_Collections_Generic_IEnumerable_1_Xunit_Sdk_IXunitTestCase_Xunit_Abstractions_IMessageSink_Xunit_Abstractions_ITestFrameworkExecutionOptions (wasm://wasm/0b9a6656:wasm-function[71762]:0x10eef2c)
    at xunit_execution_dotnet_Xunit_Sdk_TestFrameworkExecutor_1_TTestCase_REF_RunTests_System_Collections_Generic_IEnumerable_1_Xunit_Abstractions_ITestCase_Xunit_Abstractions_IMessageSink_Xunit_Abstractions_ITestFrameworkExecutionOptions (wasm://wasm/0b9a6656:wasm-function[71639]:0x10e1451)
    at xunit_runner_utility_netcoreapp10_Xunit_Xunit2_RunTests_System_Collections_Generic_IEnumerable_1_Xunit_Abstractions_ITestCase_Xunit_Abstractions_IMessageSink_Xunit_Abstractions_ITestFrameworkExecutionOptions (wasm://wasm/0b9a6656:wasm-function[74745]:0x1192c92)
    at xunit_runner_utility_netcoreapp10_TestFrameworkExtensions_RunTests_Xunit_Abstractions_ITestFrameworkExecutor_System_Collections_Generic_IEnumerable_1_Xunit_Abstractions_ITestCase_Xunit_IMessageSinkWithTypes_Xunit_Abstractions_ITestFrameworkExecutionOptions (wasm://wasm/0b9a6656:wasm-function[74474]:0x118502f)
    at invoke_viiiii (./dotnet.js:8188:27)
    at Microsoft_DotNet_XHarness_TestRunners_Xunit_Microsoft_DotNet_XHarness_TestRunners_Xunit_ThreadlessXunitTestRunner__Rund__6_MoveNext (wasm://wasm/0b9a6656:wasm-function[3355]:0x132a07)
    at Microsoft_DotNet_XHarness_TestRunners_Xunit_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Microsoft_DotNet_XHarness_TestRunners_Xunit_ThreadlessXunitTestRunner__Rund__6_Microsoft_DotNet_XHarness_TestRunners_Xunit_ThreadlessXunitTestRunner__Rund__6_ (wasm://wasm/0b9a6656:wasm-function[3554]:0x1403af)
    at Microsoft_DotNet_XHarness_TestRunners_Xunit_Microsoft_DotNet_XHarness_TestRunners_Xunit_ThreadlessXunitTestRunner_Run_System_Collections_Generic_IEnumerable_1_Microsoft_DotNet_XHarness_TestRunners_Common_TestAssemblyInfo (wasm://wasm/0b9a6656:wasm-function[3336]:0x12db4e)
    at invoke_iiii (./dotnet.js:8155:34)
    at Microsoft_DotNet_XHarness_TestRunners_Common_Microsoft_DotNet_XHarness_TestRunners_Common_ApplicationEntryPoint__InternalRunAsyncd__38_MoveNext (wasm://wasm/0b9a6656:wasm-function[2661]:0x10c6c7)
    at Microsoft_DotNet_XHarness_TestRunners_Common_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Microsoft_DotNet_XHarness_TestRunners_Common_ApplicationEntryPoint__InternalRunAsyncd__38_Microsoft_DotNet_XHarness_TestRunners_Common_ApplicationEntryPoint__InternalRunAsyncd__38_ (wasm://wasm/0b9a6656:wasm-function[2829]:0x117c48)
    at Microsoft_DotNet_XHarness_TestRunners_Common_Microsoft_DotNet_XHarness_TestRunners_Common_ApplicationEntryPoint_InternalRunAsync_Microsoft_DotNet_XHarness_TestRunners_Common_LogWriter (wasm://wasm/0b9a6656:wasm-function[2652]:0x10a021)
    at invoke_iiii (./dotnet.js:8155:34)
    at Microsoft_DotNet_XHarness_TestRunners_Common_Microsoft_DotNet_XHarness_TestRunners_Common_ApplicationEntryPoint__InternalRunAsyncd__39_MoveNext (wasm://wasm/0b9a6656:wasm-function[2669]:0x10d8b7)
    at Microsoft_DotNet_XHarness_TestRunners_Common_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Microsoft_DotNet_XHarness_TestRunners_Common_ApplicationEntryPoint__InternalRunAsyncd__39_Microsoft_DotNet_XHarness_TestRunners_Common_ApplicationEntryPoint__InternalRunAsyncd__39_ (wasm://wasm/0b9a6656:wasm-function[2831]:0x117f82)
    at Microsoft_DotNet_XHarness_TestRunners_Common_Microsoft_DotNet_XHarness_TestRunners_Common_ApplicationEntryPoint_InternalRunAsync_Microsoft_DotNet_XHarness_TestRunners_Common_ApplicationOptions_System_IO_TextWriter_System_IO_TextWriter (wasm://wasm/0b9a6656:wasm-function[2653]:0x10a170)
    at invoke_iiiiii (./dotnet.js:8210:34)
    at Microsoft_DotNet_XHarness_TestRunners_Common_Microsoft_DotNet_XHarness_TestRunners_Common_WasmApplicationEntryPointBase__RunAsyncd__4_MoveNext (wasm://wasm/0b9a6656:wasm-function[2762]:0x1132af)
    at Microsoft_DotNet_XHarness_TestRunners_Common_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Microsoft_DotNet_XHarness_TestRunners_Common_WasmApplicationEntryPointBase__RunAsyncd__4_Microsoft_DotNet_XHarness_TestRunners_Common_WasmApplicationEntryPointBase__RunAsyncd__4_ (wasm://wasm/0b9a6656:wasm-function[2927]:0x11af08)
    at Microsoft_DotNet_XHarness_TestRunners_Common_Microsoft_DotNet_XHarness_TestRunners_Common_WasmApplicationEntryPointBase_RunAsync (wasm://wasm/0b9a6656:wasm-function[2757]:0x11300e)
    at invoke_iii (./dotnet.js:8122:34)
    at Microsoft_DotNet_XHarness_TestRunners_Xunit_Microsoft_DotNet_XHarness_TestRunners_Xunit_WasmApplicationEntryPoint__Rund__28_MoveNext (wasm://wasm/0b9a6656:wasm-function[3380]:0x1358e8)
    at Microsoft_DotNet_XHarness_TestRunners_Xunit_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_Microsoft_DotNet_XHarness_TestRunners_Xunit_WasmApplicationEntryPoint__Rund__28_Microsoft_DotNet_XHarness_TestRunners_Xunit_WasmApplicationEntryPoint__Rund__28_ (wasm://wasm/0b9a6656:wasm-function[3660]:0x143282)
    at Microsoft_DotNet_XHarness_TestRunners_Xunit_Microsoft_DotNet_XHarness_TestRunners_Xunit_WasmApplicationEntryPoint_Run (wasm://wasm/0b9a6656:wasm-function[3378]:0x135467)
    at invoke_iii (./dotnet.js:8122:34)
    at WasmTestRunner_SimpleWasmTestRunner__Maind__0_MoveNext (wasm://wasm/0b9a6656:wasm-function[69033]:0x1017ad0)
    at WasmTestRunner_System_Runtime_CompilerServices_AsyncMethodBuilderCore_Start_SimpleWasmTestRunner__Maind__0_SimpleWasmTestRunner__Maind__0_ (wasm://wasm/0b9a6656:wasm-function[69087]:0x101b9f9)
    at WasmTestRunner_SimpleWasmTestRunner_Main_string__ (wasm://wasm/0b9a6656:wasm-function[69030]:0x1017023)
    at corlib_aot_wrapper_gsharedvt_out_sig_pinvoke_obj_obj (wasm://wasm/0b9a6656:wasm-function[53039]:0xcec9d2)
    at jit_call_cb (wasm://wasm/0b9a6656:wasm-function[78848]:0x1282619)
    at invoke_vi (./dotnet.js:8199:27)
    at mono_llvm_cpp_catch_exception (wasm://wasm/0b9a6656:wasm-function[79279]:0x12a101b)
    at do_jit_call (wasm://wasm/0b9a6656:wasm-function[78795]:0x128140b)
    at interp_exec_method (wasm://wasm/0b9a6656:wasm-function[78723]:0x1273b58)
    at interp_runtime_invoke (wasm://wasm/0b9a6656:wasm-function[78722]:0x1272649)
    at mono_jit_runtime_invoke (wasm://wasm/0b9a6656:wasm-function[95083]:0x1455382)
    at do_runtime_invoke (wasm://wasm/0b9a6656:wasm-function[83692]:0x1312f55)
    at mono_runtime_try_invoke (wasm://wasm/0b9a6656:wasm-function[83703]:0x1313661)
    at mono_runtime_invoke (wasm://wasm/0b9a6656:wasm-function[83780]:0x1315733)
    at mono_wasm_invoke_method (wasm://wasm/0b9a6656:wasm-function[97052]:0x1494f22)
    at Module._mono_wasm_invoke_method (./dotnet.js:7395:117)
    at Object.o.<computed> (./dotnet.js:3:4100)
    at _call_method_with_converted_args (./dotnet.js:3:49824)
    at call_method (./dotnet.js:3:48942)
    at ./dotnet.js:3:50765
    at mono_call_assembly_entry_point (./dotnet.js:3:50893)
    at Object.mono_run_main (./dotnet.js:3:59443)
    at Object.init (test-main.js:229:43)
    at Object.onDotnetReady (test-main.js:173:17)
    at finalize_startup (./dotnet.js:3:64360)
    at mono_wasm_after_runtime_initialized (./dotnet.js:3:61254)
    at callRuntimeCallbacks (./dotnet.js:1533:11)
    at postRun (./dotnet.js:1235:3)
    at doRun (./dotnet.js:8927:5)
    at run (./dotnet.js:8940:5)
    at runCaller (./dotnet.js:8894:19)
    at removeRunDependency (./dotnet.js:1304:7)
    at receiveInstance (./dotnet.js:1445:5)
    at receiveInstantiationResult (./dotnet.js:1456:5)
Process v8.cmd exited with 1
```

radical commented:
Earlier in the build:
```
EXEC : warning : undefined symbol: htons (referenced by $inetPton6__deps: ['htons','$jstoi_q'], referenced by $writeSockaddr__deps: ['$Sockets','$inetPton4','$inetPton6','$zeroMemory'], referenced by __syscall_accept4__deps: ['$getSocketFromFD','$writeSockaddr','$DNS','$SYSCALLS'], referenced by top-level compiled C/C++ code) [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
EXEC : warning : undefined symbol: ntohs (referenced by $inetNtop6__deps: ['$inetNtop4','ntohs'], referenced by $readSockaddr__deps: ['$Sockets','$inetNtop4','$inetNtop6','ntohs'], referenced by $getSocketAddress__deps: ['$readSockaddr','$FS','$DNS'], referenced by __syscall_bind__deps: ['$getSocketFromFD','$getSocketAddress','$SYSCALLS'], referenced by top-level compiled C/C++ code) [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
EXEC : warning : undefined symbol: _get_daylight (referenced by tzset_impl__deps: ['_get_daylight','_get_timezone','_get_tzname'], referenced by tzset__deps: ['tzset_impl'], referenced by localtime_r__deps: ['tzset'], referenced by top-level compiled C/C++ code) [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
EXEC : warning : undefined symbol: _get_timezone (referenced by tzset_impl__deps: ['_get_daylight','_get_timezone','_get_tzname'], referenced by tzset__deps: ['tzset_impl'], referenced by localtime_r__deps: ['tzset'], referenced by top-level compiled C/C++ code) [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
EXEC : warning : undefined symbol: _get_tzname (referenced by tzset_impl__deps: ['_get_daylight','_get_timezone','_get_tzname'], referenced by tzset__deps: ['tzset_impl'], referenced by localtime_r__deps: ['tzset'], referenced by top-level compiled C/C++ code) [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
   "C:/helix/work/correlation/build/emsdk/upstream\bin\wasm-opt" --post-emscripten -Oz --low-memory-unused --zero-filled-memory --strip-producers 
```

radical committed:
Merge branch 'main' into pr-emscripten-311-bump

radical committed:
[wasm] Export more functions

`htons,ntohs,_get_daylight,_get_timezone,_get_tzname`

.. to fix:

```
EXEC : warning : undefined symbol: htons (referenced by $inetPton6__deps: ['htons','$jstoi_q'], referenced by $writeSockaddr__deps: ['$Sockets','$inetPton4','$inetPton6','$zeroMemory'], referenced by __syscall_accept4__deps: ['$getSocketFromFD','$writeSockaddr','$DNS','$SYSCALLS'], referenced by top-level compiled C/C++ code) [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
EXEC : warning : undefined symbol: ntohs (referenced by $inetNtop6__deps: ['$inetNtop4','ntohs'], referenced by $readSockaddr__deps: ['$Sockets','$inetNtop4','$inetNtop6','ntohs'], referenced by $getSocketAddress__deps: ['$readSockaddr','$FS','$DNS'], referenced by __syscall_bind__deps: ['$getSocketFromFD','$getSocketAddress','$SYSCALLS'], referenced by top-level compiled C/C++ code) [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
EXEC : warning : undefined symbol: _get_daylight (referenced by tzset_impl__deps: ['_get_daylight','_get_timezone','_get_tzname'], referenced by tzset__deps: ['tzset_impl'], referenced by localtime_r__deps: ['tzset'], referenced by top-level compiled C/C++ code) [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
EXEC : warning : undefined symbol: _get_timezone (referenced by tzset_impl__deps: ['_get_daylight','_get_timezone','_get_tzname'], referenced by tzset__deps: ['tzset_impl'], referenced by localtime_r__deps: ['tzset'], referenced by top-level compiled C/C++ code) [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
EXEC : warning : undefined symbol: _get_tzname (referenced by tzset_impl__deps: ['_get_daylight','_get_timezone','_get_tzname'], referenced by tzset__deps: ['tzset_impl'], referenced by localtime_r__deps: ['tzset'], referenced by top-level compiled C/C++ code) [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
```

```
Starting:    Microsoft.Extensions.Logging.Tests.dll
Message1
:
testKey3:testValue, testKey2:, testKey1:
missing function: ntohs
Error
    at Object.onAbort (test-main.js:176:60)
    at abort (./dotnet.js:1316:24)
    at _ntohs (./dotnet.js:4990:35)
    at readSockaddr (./dotnet.js:5091:18)
    ...
```

radical committed:
fix exported function names

radical commented:
I added these functions to the `EXPORTED_FUNCTIONS` list. Feel free to remove if that doesn't look correct.

radical commented:
/azp run runtime-wasm

radekdoulik commented:
> I added these functions to the `EXPORTED_FUNCTIONS` list. Feel free to remove if that doesn't look correct.

Yes, these are still needed in 3.1.1. The TZ deps were fixed in 3.1.3, so we can try to remove them in following emscripten bumps.

radekdoulik commented:
The above mentioned PRs are to be merged together with this one to stay on 3.1.1 with the following darc flow updates.
